core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.kt

```diff
@@ -0,0 +1,141 @@
+import android.widget.Switch
+import androidx.appcompat.widget.SearchView
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import io.reactivex.disposables.CompositeDisposable
+import kotlinx.android.synthetic.main.activity_history.history_switch
+import kotlinx.android.synthetic.main.activity_history.recycler_view
+import kotlinx.android.synthetic.main.layout_toolbar.toolbar
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R.id
+import org.kiwix.kiwixmobile.core.R.string
+import org.kiwix.kiwixmobile.core.R2.id.menu_history_search
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.coreActivityComponent
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.viewModel
+import org.kiwix.kiwixmobile.core.history.HistoryAdapter.OnItemClickListener
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryAdapter2
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryDelegate.HistoryItemDelegate
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem.HistoryItem
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.Filter
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.OnItemLongClick
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.OnSwitch
+import org.kiwix.kiwixmobile.core.history.viewmodel.HistoryViewModel
+import org.kiwix.kiwixmobile.core.history.viewmodel.State
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.NoResults
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.Results
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.SelectionResults
+import org.kiwix.kiwixmobile.core.utils.SimpleTextListener
+import java.util.ArrayList
+import javax.inject.Inject
+
+class HistoryActivity : OnItemClickListener, BaseActivity() {
+
+  val activityComponent by lazy { coreActivityComponent }
+  private val historyList: List<HistoryListItem> = ArrayList()
+  private val fullHistory: List<HistoryListItem> = ArrayList()
+  val search = menu_history_search
+  private val deleteList: List<HistoryListItem> = ArrayList()
+  @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+  private lateinit var recyclerView: RecyclerView
+  private lateinit var historySwitch: Switch
+  private val historyViewModel by lazy { viewModel<HistoryViewModel>(viewModelFactory) }
+  private val compositeDisposable = CompositeDisposable()
+
+  private val historyAdapter: HistoryAdapter2 by lazy {
+    HistoryAdapter2(
+      HistoryItemDelegate(deleteList, this)
+    )
+  }
+
+  override fun injection(coreComponent: CoreComponent) {
+    coreComponent.activityComponentBuilder()
```

Comment:
you should only ever call `inject(this)` once per class so you can delete using the builder, the extension property `activityComponent` already does that for you